### ISSUE TITLE
podman seems to ignore certs.d directories when running rootless.

### ISSUE THREAD

**Issue by cesan3** [containers/libpod#2726]
/kind bug

**Description**

When running podman pull/push rootless, I'm forced to pass the --cert-dir option to provide the certificates (tried the defaults  /etc/containers/certs.d/<certs>, or /etc/containers/<host>/<certs>, or even /etc/docker/certs.d/<host>/<certs>.

I'm ok, passing the --cert-dir option, for pull/push, but, for podman image sign, I don't have the option, and I get the complaint from the 

**Steps to reproduce the issue:**

1. setup a secure docker registry with tls (using my certs)
2. as an unprivileged user, build an image
3. tag the image and push it to the registry
* podman push --cert-dir <path-to-my-certs>
4. try to sign the image with your gpg key
* podman image sign --sign-by <my-gpg-key> docker://<registry:image:tag>
**Describe the results you received:**

podman produces error:

error getting manifest: pinging docker registry returned: Get https://<registry:<port>/v2/: x509: certificate signed by unknown authority

**Describe the results you expected:**

Either, podman image sing allows passing the --cert-dir option, or the /etc/containers/certs.d/ directory is honored.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:         1.0.0
Go Version:    go1.10.2
OS/Arch:        linux/amd64
```

**Output of `podman info --debug`:**

```
debug::
  compiler: gc
  git commit: ""
  go version: go1.10.2
  podman versionL: 1.0.0
host:
  BuildahVersion: 1.6-dev
  Conmon:
    package: podman-1.0.0-1.git82e8011.el7.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.14.0-dev, commit: 635328ab9b2058e40335533f35d595604666f1e4-dirty'
  Distribution:
    distribution: '"rhel"'
    version: "7.6"
   MemFree: 686776320
   MemTotal: 8201482240
    OCIRuntime:
      package:  runc-1.0.0-59.dev.git2abd837.el7.x86_64
      path: /usr/bin/runc
      version: 'runc version spec: 1.0.0'
    SwapFree: 930344960
    SwapTotal: 1073737728
    arch: amd64
    cpus: 4
    hostname: my-server.test.com
    kernel: 3.10.0-957.5.1.el7.x86_64
    os: linux
    rootless: true
    uptime: 478h 38m 58.07s (Approximately 19.92 days)
insecure registries:
  registries:
    - my-insecure-registry.test.com:5000
registries:
  registries:
    - registry.access.redhat.com
    - docker.io
    - registry.fedoraproject.org
    - quay.io
    - registry.centos.org
store:
  ConfigFile: /home/my_user/.config/containers/storage.conf
  ContainerStore:
    number: 75
  GraphDriverName: vfs
  GraphOptions: null
  GraphRoot: /home/my_user/.local/share/containers/storage
  GraphStatus: {}
  ImageStore:
    number: 10
  RunRoot: /run/user/1000
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Vmware guest

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by rhatdan** [comment]
@vrothberg WDYT?

**Comment by zdzichu** [comment]
Hey, any comments? I have exactly same problem - I have CA certs used to sign registry's cert in my system database, and additionally in `/etc/containers/certs.d/<hostname>/*.crt`. With either `strace` or `podman --log-level debug` I see podman is reading those, but login fails unless I pass `--tls-verify=false`. The error is

```
Error: error authenticating creds for "registry.myhost.local": pinging docker registry returned: Get https://registry.myhost.local/v2/: EOF
```

Any hints how to debug this further?

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.12.7
  podman version: 1.4.4
host:
  BuildahVersion: 1.9.0
  Conmon:
    package: podman-1.4.4-4.fc30.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.0.0-dev, commit: 164df8af4e62dc759c312eab4b97ea9fb6b5f1fc'
  Distribution:
    distribution: fedora
    version: "30"
```

**Comment by vrothberg** [comment]
@cesan3, thanks a lot for opening the issue and apologies for the long silence. This one slipped through.

> I'm ok, passing the --cert-dir option, for pull/push, but, for podman image sign, I don't have the option [...]

@QiWang19 can you have a look? Any reason we didn't add the option? (Cc @rhatdan)

> When running podman pull/push rootless, I'm forced to pass the --cert-dir option to provide the certificates (tried the defaults /etc/containers/certs.d/, or /etc/containers//, or even /etc/docker/certs.d//.

@giuseppe, have you seen this before?

**Comment by vrothberg** [comment]
@zdzichu, thanks for the ping. We're looking into it now.

**Comment by QiWang19** [comment]
There's no specific reason. I just didn't take `--cert-dir` into account. Do I need to add `--cert-dir`?

**Comment by vrothberg** [comment]
> There's no specific reason. I just didn't take `--cert-dir` into account. Do I need to add `--cert-dir`?

Yes, I think we to support `--cert-dir`.

**Comment by TomSweeneyRedHat** [comment]
Needing to pass the --cert-dir when in rootless mode and the certs are in the default location seems like a bug that needs to be chased down.  I'm fine with adding --cert-dir to the `podman image sign` command, but it to shouldn't be necessary for the default case.

**Comment by giuseppe** [comment]
> > 
> 
> @giuseppe, have you seen this before?

AFAIK, there is no difference between rootless and root for the cert.d location, could you confirm that it works when running as root?  Could you attach the output with `--log-level debug`?

**Comment by QiWang19** [comment]
The above error `podman produces error: error getting manifest:` was returned by `GetManifest()`. I don't think the sign --cert-dir can avoid this error.

**Comment by rhatdan** [comment]
@QiWang19 I don't understand, are you saying this can not be done?

**Comment by QiWang19** [comment]
> @QiWang19 I don't understand, are you saying this can not be done?
This can be done. Created #3833 PR for --cert-dir

**Comment by baude** [comment]
we believe this issue is now closed with PR #3833 .  please re-open if you find that not to be true.
